kernel/kernel-api/src/main/java/io/delta/kernel/internal/ReplaceTableTransactionBuilderImpl.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal;
+
+import static io.delta.kernel.internal.DeltaErrors.requireSchemaForReplaceTable;
+
+import io.delta.kernel.Operation;
+import io.delta.kernel.Transaction;
+import io.delta.kernel.engine.Engine;
+import io.delta.kernel.exceptions.TableNotFoundException;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+public class ReplaceTableTransactionBuilderImpl extends TransactionBuilderImpl {
+
+  public ReplaceTableTransactionBuilderImpl(TableImpl table, String engineInfo) {
+    super(table, engineInfo, Operation.REPLACE_TABLE);
+  }
+
+  @Override
+  public Transaction build(Engine engine) {
+    try {
+      withMaxRetries(0); // We don't support conflict resolution yet so disable retries for now
```

Comment:
So the user can set `myBuilder.withMaxRetries(10)` and then we will just ignore it? Would that be weird or confusing? Should we override that method and either fail or log a warning?

Reply:
I'm open to opinions on this one but I think it's okay to retry _less_ than what is specified (what we do if we find a conflict that we cannot resolve anyways). It's called "max" retries too :)

Also this is similar to what would happen if for example we haven't updated our conflict resolution logic yet for some operation to be more fine-grained. We always might fail conflict resolution for conflicting commits which in some way might be resolvable. This just basically means we always fail conflict resolution for replace at the moment.